src/sdk/namespace_fs.js

```diff
@@ -924,6 +919,24 @@ class NamespaceFS {
         }
     }
 
+    async validate_results(results, fs_context) {
+        await Promise.all(results.map(async r => {
+            if (r.common_prefix) return;
+            const entry_path = path.join(this.bucket_path, r.key);
+            //If entry is outside of bucket, returns stat of symbolic link
+            const use_lstat = !(await this._is_path_in_bucket_boundaries(fs_context, entry_path));
+            try {
+                r.stat = await nb_native().fs.stat(fs_context, entry_path, { use_lstat });
+            } catch (err) {
```

Comment:
notice that you throw for all errors and not just EACCES( permission denied) is this intentional? maybe it would be better to create a separate function that doesn't throw for selected errors. like we do in unlink_ignore_enoent. that way we can reuse in case we would like to handle permission issues in other places. see [stat](https://man7.org/linux/man-pages/man2/stat.2.html) for possible error list

Reply:
also that way we will be able to handle what happens in case of EACCES outside the catch. something like:
```
const stat = stat_ignore_eaccess();
if (!stat) {
    ....
}

Reply:
@nadavMiz Here only error we will get is an access error, files are added to the result array only because they already exist in the folder. But I like the idea of util method for access check.